### PR TITLE
fix(Props/Dialog/Button): Prop Fixes for Button / Dialog

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

A few assorted fixes for increased prop flexibility and function fixes for Dialog and Button.

 **Props**
- Adds `flexItem` and `gridItem` to standard `layout` props allowing specification of grid / flex behavior for most components without direct control over all grid and flex props.  This lets primitives specify their own behavior within any grid or flex component without needing a separate container.
- Exposes separate `margin` and `padding` system prop subgroups to be used more surgically.   This is to fix a an issue with button being unable to have `padding` without breaking component behavior (ButtonOutline depends on specific padding). 

**Button**
- Adds `layout` `positioning` and `margin` as system props to all ButtonComponents.
- Also fixes issues with possible prop collisions with `HTMLButton` intrinsic attributes by filtering probably collisions out.

**Dialog**
- Fixes an issue where the width was removed in the recent upgrade, but adds new sized variants of medium and small.
- Removes unnecessary layout components by setting grid area directly on components in those areas.

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/gamut/src/Dialog/index.tsx [hunk 1]
```diff
@@ -1,24 +1,35 @@
...
 
-const ModalWrapper = styled.div(
+const ModalWrapper = styled.div<SizeVariants>(
   system.css({
     position: 'relative',
     zIndex: 1,
     maxWidth: 'calc(100vw - 2rem)',
     borderRadius: '2px',
-  })
+    display: 'grid',
```
Comment (codecaaron) [inline packages/gamut/src/Dialog/index.tsx:30]: This ensures `ModalBody` will always be the [size of its container](https://github.com/Codecademy/client-modules/pull/1691/files#diff-b4bd9d1f3f6e11ad4a35de440f038de5e68a96fe15d349d16bf1b040a99b879bR18)

File: packages/gamut/src/Dialog/index.tsx [hunk 2]
```diff
@@ -44,31 +55,16 @@ const ModalBody = styled.div(
     rowGap: 12,
     columnGap: 16,
     gridTemplateColumns: '1fr min-content 2rem',
-    gridTemplateRows: 'repeact(3, auto)',
```
Comment (codecaaron) [inline packages/gamut/src/Dialog/index.tsx:47]: This was actually broken, but did not change the behavior.

File: packages/gamut/src/Dialog/index.tsx [hunk 3]
```diff
@@ -44,31 +55,16 @@ const ModalBody = styled.div(
...
+    gridTemplateRows: 'max-content 1fr max-content',
```
Comment (codecaaron) [inline packages/gamut/src/Dialog/index.tsx:58]: This will probably need to be updated in the future as we use these modals more.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.11-alpha.c644b0.0<br/>@codecademy/gamut-kit@0.3.65-alpha.c644b0.0<br/>@codecademy/gamut-labs@15.2.5-alpha.c644b0.0<br/>@codecademy/gamut-styles@9.7.1-alpha.c644b0.0<br/>@codecademy/gamut-tests@2.3.23-alpha.c644b0.0<br/>@codecademy/gamut@32.7.2-alpha.c644b0.0<br/>@codecademy/konamimojisplosion@0.2.23-alpha.c644b0.0<br/>@codecademy/markdown-overrides@0.5.35-alpha.c644b0.0<br/>@codecademy/styleguide@36.9.2-alpha.c644b0.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608b735b7e9cad1a6daa6c92--gamut-preview.netlify.app>https://608b735b7e9cad1a6daa6c92--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608b735b7e9cad1a6daa6c92>Deploy Logs</a></p>

File: packages/gamut/src/Dialog/index.tsx [hunk 4]
```diff
@@ -89,12 +85,9 @@ export const Dialog: React.FC<DialogProps> = ({
   confirmCta,
   cancelCta,
   onRequestClose,
+  size,
```
Comment (slin12) [inline packages/gamut/src/Dialog/index.tsx:88]: is size required here? if not, what are we defaulting to?

**Review by slin12 — APPROVED** [review]

File: packages/gamut/src/Dialog/index.tsx (hunk 4 above)
Comment (codecaaron) [reply packages/gamut/src/Dialog/index.tsx:88]: Ah good catch I removed the default variant but forgot to add it as a default prop <3.  Fixing!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.11-alpha.0418d0.0<br/>@codecademy/gamut-kit@0.3.65-alpha.0418d0.0<br/>@codecademy/gamut-labs@15.2.5-alpha.0418d0.0<br/>@codecademy/gamut-styles@9.7.1-alpha.0418d0.0<br/>@codecademy/gamut-tests@2.3.23-alpha.0418d0.0<br/>@codecademy/gamut@32.7.2-alpha.0418d0.0<br/>@codecademy/konamimojisplosion@0.2.23-alpha.0418d0.0<br/>@codecademy/markdown-overrides@0.5.35-alpha.0418d0.0<br/>@codecademy/styleguide@36.9.2-alpha.0418d0.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608c1d88f81ef92f376194cb--gamut-preview.netlify.app>https://608c1d88f81ef92f376194cb--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608c1d88f81ef92f376194cb>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.11.1-alpha.203d02.0<br/>@codecademy/gamut-kit@0.3.66-alpha.203d02.0<br/>@codecademy/gamut-labs@15.2.6-alpha.203d02.0<br/>@codecademy/gamut-styles@9.7.1-alpha.203d02.0<br/>@codecademy/gamut-tests@2.3.23-alpha.203d02.0<br/>@codecademy/gamut@32.7.3-alpha.203d02.0<br/>@codecademy/konamimojisplosion@0.2.23-alpha.203d02.0<br/>@codecademy/markdown-overrides@0.5.35-alpha.203d02.0<br/>@codecademy/styleguide@36.9.3-alpha.203d02.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608c20484f893c42264760f0--gamut-preview.netlify.app>https://608c20484f893c42264760f0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608c20484f893c42264760f0>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.11.2-alpha.05cfa9.0<br/>@codecademy/gamut-kit@0.3.67-alpha.05cfa9.0<br/>@codecademy/gamut-labs@15.2.7-alpha.05cfa9.0<br/>@codecademy/gamut-styles@9.7.2-alpha.05cfa9.0<br/>@codecademy/gamut-tests@2.3.24-alpha.05cfa9.0<br/>@codecademy/gamut@32.7.4-alpha.05cfa9.0<br/>@codecademy/konamimojisplosion@0.2.24-alpha.05cfa9.0<br/>@codecademy/markdown-overrides@0.5.36-alpha.05cfa9.0<br/>@codecademy/styleguide@36.9.4-alpha.05cfa9.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608c71b69328aa0e31e95a38--gamut-preview.netlify.app>https://608c71b69328aa0e31e95a38--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608c71b69328aa0e31e95a38>Deploy Logs</a></p>

**Review by JoshuaKGoldberg — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.11.2-alpha.f7bf9f.0<br/>@codecademy/gamut-kit@0.3.67-alpha.f7bf9f.0<br/>@codecademy/gamut-labs@15.2.7-alpha.f7bf9f.0<br/>@codecademy/gamut-styles@9.7.2-alpha.f7bf9f.0<br/>@codecademy/gamut-tests@2.3.24-alpha.f7bf9f.0<br/>@codecademy/gamut@32.7.4-alpha.f7bf9f.0<br/>@codecademy/konamimojisplosion@0.2.24-alpha.f7bf9f.0<br/>@codecademy/markdown-overrides@0.5.36-alpha.f7bf9f.0<br/>@codecademy/styleguide@36.9.4-alpha.f7bf9f.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.11.2-alpha.c50559.0<br/>@codecademy/gamut-kit@0.3.67-alpha.c50559.0<br/>@codecademy/gamut-labs@15.2.7-alpha.c50559.0<br/>@codecademy/gamut-styles@9.7.2-alpha.c50559.0<br/>@codecademy/gamut-tests@2.3.24-alpha.c50559.0<br/>@codecademy/gamut@32.7.4-alpha.c50559.0<br/>@codecademy/konamimojisplosion@0.2.24-alpha.c50559.0<br/>@codecademy/markdown-overrides@0.5.36-alpha.c50559.0<br/>@codecademy/styleguide@36.9.4-alpha.c50559.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608f4bcfc41e77ab85a5a195--gamut-preview.netlify.app>https://608f4bcfc41e77ab85a5a195--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608f4bcfc41e77ab85a5a195>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.11.2-alpha.ab90e2.0<br/>@codecademy/gamut-kit@0.3.67-alpha.ab90e2.0<br/>@codecademy/gamut-labs@15.2.7-alpha.ab90e2.0<br/>@codecademy/gamut-styles@9.7.2-alpha.ab90e2.0<br/>@codecademy/gamut-tests@2.3.24-alpha.ab90e2.0<br/>@codecademy/gamut@32.7.4-alpha.ab90e2.0<br/>@codecademy/konamimojisplosion@0.2.24-alpha.ab90e2.0<br/>@codecademy/markdown-overrides@0.5.36-alpha.ab90e2.0<br/>@codecademy/styleguide@36.9.4-alpha.ab90e2.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609017a64ef0540e21472a5e--gamut-preview.netlify.app>https://609017a64ef0540e21472a5e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609017a64ef0540e21472a5e>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]
looks great! only thing is icon buttons seem to be broken on both ColorMode previews: 
image

**Comment by codecaaron** [comment]
> looks great! only thing is icon buttons seem to be broken on both ColorMode previews:
> image

I think this is isolated to he button component stories, will fix.

The other ktichen sink is working here: https://609017a64ef0540e21472a5e--gamut-preview.netlify.app/storybook/?path=/docs/foundations-colormode--page

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.13.2-alpha.493787.0<br/>@codecademy/gamut-kit@0.3.70-alpha.493787.0<br/>@codecademy/gamut-labs@15.2.10-alpha.493787.0<br/>@codecademy/gamut-styles@9.7.3-alpha.493787.0<br/>@codecademy/gamut-tests@2.3.25-alpha.493787.0<br/>@codecademy/gamut@32.8.3-alpha.493787.0<br/>@codecademy/konamimojisplosion@0.2.25-alpha.493787.0<br/>@codecademy/markdown-overrides@0.5.37-alpha.493787.0<br/>@codecademy/styleguide@36.11.2-alpha.493787.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609060a5525cbd00bab0ada6--gamut-preview.netlify.app>https://609060a5525cbd00bab0ada6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609060a5525cbd00bab0ada6>Deploy Logs</a></p>

**Pull request merged**